### PR TITLE
Swipe

### PR DESCRIPTION
**Pull request by brentdur**
1. Adds Swipe to Checkin
2. Allows both swipe and the older style
3. If the account doesn't have an NYU id number it dynamically adds it or creates an account
4. Supports T@N API V3

### CODE REVIEW

**Comment by brentdur** [comment]
Is currently configured to use **TEST** Api deployment

**Comment by AbhiAgarwal** [comment]
v2 branch pushes automatically straight to checkin.techatnyu.org. I can setup a staging environment for it instead. Can you merge it into a develop branch instead? I'll create the branch. You could also just change it to `v3` instead of `v3-test`.

File: app/js/controllers/entry-ctrl.js [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+					rsvp_person($scope.person.id);
+				});
+			}
+			else {
+				elseCallback()
+			}
+		});
+	}
+
+	function rsvp_person(id) {
```
Comment (AbhiAgarwal) [inline app/js/controllers/entry-ctrl.js]: Can we do consistent camelCase

File: app/js/controllers/entry-ctrl.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+				.then(function(post){
+					$scope.rsvpd = true;
+					resetTimeout();
+				});
+			}
+		});
+	}
+
+	//updates the person's nNumber record
+	function update_person(number) {
```
Comment (AbhiAgarwal) [inline app/js/controllers/entry-ctrl.js]: Can we do consistent camelCase

File: app/js/controllers/entry-ctrl.js [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+				if (rsvpIds[val.id] === true){
+					allPeople.push(val.id);
+					$scope.idToPerson[val.id] = val.attributes.name;
+				}
+			}).value();
+		});
+	});
+
+	//autocomplete suggestions
+	function suggest_state(term) {
```
Comment (AbhiAgarwal) [inline app/js/controllers/entry-ctrl.js]: Can we do consistent camelCase

**Comment by AbhiAgarwal** [comment]
Otherwise it looks pretty good! I'm excited to test it out. I ordered my card reader earlier. 

q: where does the magnetic reader thing run? Is there a separate app somewhere?

**Comment by brentdur** [comment]
@AbhiAgarwal The card reader just emulates keyboard entry. So when you swipe it just enters in text it sees into whatever text field has focus. It then follows the data with "Enter". It's the same as if it was a usb keyboard

**Comment by AbhiAgarwal** [comment]
@brentdur okay! that's actually pretty perfect. I'll try it out during the week!

**Pull request closed without merging**